File: src/main/java/github/macro/build_info/gems/UpdateGem.java
package github.macro.build_info.gems;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by Macro303 on 2020-Jan-08.
 */
public class UpdateGem {
	@Nullable
	private final Gem oldGem;
	@Nullable
	private final Gem newGem;
	@NotNull
	private final String reason;

	public UpdateGem(@Nullable Gem oldGem, @Nullable Gem newGem, @NotNull String reason) {
		this.oldGem = oldGem;
		this.newGem = newGem;
		this.reason = reason;
	}

	@Nullable
	public Gem getOldGem() {
		return oldGem;
	}

	@Nullable
	public Gem getNewGem() {
		return newGem;
	}

	@NotNull
	public String getReason() {
		return reason;
	}
}

File: src/main/kotlin/github/macro/build_info/gems/UpdateGem.kt
package github.macro.build_info.gems

import javafx.beans.property.SimpleObjectProperty
import javafx.beans.property.SimpleStringProperty
import tornadofx.getValue
import tornadofx.setValue

/**
 * Created by Macro303 on 2020-Jan-13.
 */
class UpdateGem(
    oldGem: Gem?,
    newGem: Gem?,
    reason: String
) {
    val oldGemProperty = SimpleObjectProperty<Gem?>()
    var oldGem by oldGemProperty

    val newGemProperty = SimpleObjectProperty<Gem?>()
    var newGem by newGemProperty

    val reasonProperty = SimpleStringProperty()
    var reason by reasonProperty

    init {
        this.oldGem = oldGem
        this.newGem = newGem
        this.reason = reason
    }

    override fun toString(): String {
        return "UpdateGem(oldGem=$oldGem, newGem=$newGem, reason=$reason)"
    }
}